gdal/gcore/gdal.h

```diff
@@ -133,10 +133,11 @@ typedef enum
     /*! Cubic B-Spline Approximation (4x4 kernel) */    GRIORA_CubicSpline = 3,
     /*! Lanczos windowed sinc interpolation (6x6 kernel) */ GRIORA_Lanczos = 4,
     /*! Average */                                      GRIORA_Average = 5,
+    /*! RMS (Quadratic Mean) */                         GRIORA_RMS = 14,
     /*! Mode (selects the value which appears most often of all the sampled points) */
                                                         GRIORA_Mode = 6,
     /*! Gauss blurring */                               GRIORA_Gauss = 7,
-    /* NOTE: values 8 to 12 are reserved for max,min,med,Q1,Q3 */
+    /* NOTE: values 8 to 13 are reserved for max,min,med,Q1,Q3,sum */
 /*! @cond Doxygen_Suppress */
                                                         GRIORA_LAST = GRIORA_Gauss
```

Comment:
Hi @rouault,

I have an issue with this. there is a check that not allow using algo upper than GRIORA_LAST, but GRIORA_RMS = 14.
the only solution that I see is to change enum values of max,min,med,Q1,Q3,sum from 8~13 to 9~14 and move rms to 8.
is there an other way to do?

Reply:
> the only solution that I see is to change enum values of max,min,med,Q1,Q3,sum

no, that would change the ABI. 
GRIORA_LAST should be updated to GRIORA_RMS.

Reply:
OK, 
should I check that checks with GRIORA_LAST should also excludes values from 8 to 13 because this values were excluded from valid values?

Reply:
> should I check that checks with GRIORA_LAST should also excludes values from 8 to 13 because this values were excluded from valid values?

that would be good

Reply:
they are not managed as valid in overview.cpp
```
GDALResampleFunction GDALGetResampleFunction( const char* pszResampling,
                                              int* pnRadius )
```

Reply:
> they are not managed as valid in overview.cpp
> 
> ```
> GDALResampleFunction GDALGetResampleFunction( const char* pszResampling,
>                                               int* pnRadius )
> ```

do you mean max,min,med,Q1,Q3,sum ? Yes, they are not valid RasterIO() resampling algorithm. Only for gdalwarp. And gdalwarp doesn't use this

Reply:
ok
I created two new values in enum GRIORA_RESERVED_START=8 and GRIORA_RESERVED_END=13 to be able correctly manage the hole in valid values

Reply:
I think it should be nice and useful to add missing resampling algorithm to RasterIO()
I could add them in other PR,
what about that @rouault ?

Reply:
> I could add them in other PR,

That could make sense